packages/roosterjs-editor-plugins/lib/plugins/TableCellSelection/mouseUtils/handleMouseDownEvent.ts

```diff
@@ -53,65 +53,99 @@ export function handleMouseDownEvent(
-            const firstTable = getTableAtCursor(editor, first);
-            const targetTable = getTableAtCursor(editor, first);
-            if (
-                firstTable! == targetTable! &&
-                safeInstanceOf(first, 'HTMLTableCellElement') &&
-                safeInstanceOf(last, 'HTMLTableCellElement')
-            ) {
-                state.vTable = new VTable(first);
-                const firstCord = getCellCoordinates(state.vTable, first);
-                const lastCord = getCellCoordinates(state.vTable, last);
-
-                if (!firstCord || !lastCord) {
-                    return;
-                }
-                state.vTable.selection = {
-                    firstCell: firstCord,
-                    lastCell: lastCord,
+                state.mouseMoveDisposer = () => {
+                    doc.removeEventListener('mouseup', mouseUpListener, true /*setCapture*/);
+                    doc.removeEventListener('mousemove', mouseMoveListener, true /*setCapture*/);
                 };
 
-                state.firstTarget = first;
-                state.lastTarget = last;
-                selectTable(editor, state);
-
-                state.tableSelection = true;
-                state.firstTable = firstTable as HTMLTableElement;
-                state.targetTable = targetTable;
-                updateSelection(editor, first, 0);
+                state.startedSelection = true;
             }
-        });
+        }
+
+        if (shiftKey) {
+            editor.runAsync(editor => {
+                const sel = editor.getDocument().defaultView?.getSelection();
+                const first = getCellAtCursor(editor, sel?.anchorNode);
+                const last = getCellAtCursor(editor, sel?.focusNode);
+                const firstTable = getTableAtCursor(editor, first);
+                const targetTable = getTableAtCursor(editor, first);
+                if (
+                    firstTable! == targetTable! &&
+                    safeInstanceOf(first, 'HTMLTableCellElement') &&
+                    safeInstanceOf(last, 'HTMLTableCellElement')
+                ) {
+                    state.vTable = new VTable(first);
+                    const firstCord = getCellCoordinates(state.vTable, first);
+                    const lastCord = getCellCoordinates(state.vTable, last);
+
+                    if (!firstCord || !lastCord) {
+                        return;
+                    }
+                    state.vTable.selection = {
+                        firstCell: firstCord,
+                        lastCell: lastCord,
+                    };
+
+                    state.firstTarget = first;
+                    state.lastTarget = last;
+                    selectTable(editor, state);
+
+                    state.tableSelection = true;
+                    state.firstTable = firstTable as HTMLTableElement;
+                    state.targetTable = targetTable;
+                    updateSelection(editor, first, 0);
```

Comment:
This part is almost the same with the previous part. Can we try to merge them?

Reply:
For example, first we always need to get selected table and cell, then coordinates, this part should be the same. Then after that we can check how many time mouse is clicked, then do different action? Not sure if this is possible.

Reply:
I implemented triple click inside of it now, by making first and last the same.